src/App/ImportUshahidiV2/Importer.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Ushahidi\App\ImportUshahidiV2;
+
+use Illuminate\Support\Collection;
+use Ushahidi\Core\Entity\Repository\EntityCreateMany; // @todo correct interface?
+use Ushahidi\App\ImportUshahidiV2;
+
+class Importer
+{
+    protected $sourceType;
+    protected $destType;
+    protected $mapper;
+    protected $mappingRepo;
+    protected $destRepo;
+
+    public function __construct(
+        string $sourceType,
+        Contracts\Mapper $mapper,
+        Contracts\ImportMappingRepository $mappingRepo,
+        EntityCreateMany $destRepo
+    ) {
+        $this->sourceType = $sourceType;
+        $this->mapper = $mapper;
+        $this->mappingRepo = $mappingRepo;
+        $this->destRepo = $destRepo;
+
+        // @todo depend on an interface that supplies getEntity?
+        $this->destType = $this->destRepo->getEntity()->getResource();
+    }
+
+    /**
+     * Run import
+     *
+     * @todo naming: run? handle? interact? process?
```

Comment:
Any suggestions for naming?

Reply:
The entire thing is namespaced under ImportUshahidiV2 - do you want that to be more obvious?

My comment above was more re: function naming not that class.